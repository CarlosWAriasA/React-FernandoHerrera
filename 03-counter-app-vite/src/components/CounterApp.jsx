import React, { useState } from "react";
import PropTypes from "prop-types";

function CounterApp({ value }) {
  const [counter, setCounter] = useState(value);

  const handleAddition = () => {
    // setCounter(counter + 1);
    return setCounter((actualValue) => {
      return actualValue + 1;
    });
  };

  const handleSubtract = () => {
    return setCounter(counter - 1);
  };

  const handleReset = () => {
    return setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button onClick={handleAddition}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button aria-label="btn-reset" onClick={handleReset}>
        Reset
      </button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
