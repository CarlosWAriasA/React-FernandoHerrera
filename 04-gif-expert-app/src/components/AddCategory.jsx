import { useState } from "react";
import PropTypes from "prop-types";
import React from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    const value = inputValue.trim();
    if (value.length <= 1) return;
    console.log(value);
    onNewCategory(value);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Search gifs..."
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
