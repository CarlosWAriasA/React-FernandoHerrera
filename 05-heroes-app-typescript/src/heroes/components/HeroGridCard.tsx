import { Heart, Eye, Zap, Brain, Gauge, Shield } from "lucide-react";

import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import { Card, CardHeader, CardContent } from "@/components/ui/card";
import { Progress } from "@/components/ui/progress";

interface Props {
  superheroName: string;
  name: string;
  universe: string;
  status: string;
  description: string;
  powers?: string;
  firstAppeard: number;
  league: string;
  image?: string;
  strength: number;
  intelligence: number;
  speed: number;
  durability: number;
  type: string;
}

export const HeroGridCard = ({
  superheroName,
  name,
  universe,
  status,
  description,
  powers,
  firstAppeard,
  league,
  image,
  strength,
  speed,
  intelligence,
  durability,
  type,
}: Props) => {
  return (
    <Card className="group overflow-hidden hover:shadow-xl transition-all duration-300 hover:-translate-y-1 bg-gradient-to-br from-white to-gray-50">
      <div className="relative h-64 overflow-hidden">
        <img
          src={image || "/placeholder.svg?height=300&width=300"}
          alt="Superman"
          className="object-cover transition-all duration-500 group-hover:scale-110"
        />

        {/* Status indicator */}
        <div className="absolute top-3 left-3 flex items-center gap-2">
          <div className="w-3 h-3 rounded-full bg-green-500" />
          <Badge
            variant="secondary"
            className="text-xs bg-white/90 text-gray-700"
          >
            {status}
          </Badge>
        </div>

        {/* Universe badge */}
        <Badge className="absolute top-3 right-3 text-xs bg-blue-600 text-white">
          {universe}
        </Badge>

        {/* Favorite button */}
        <Button
          size="sm"
          variant="ghost"
          className="absolute bottom-3 right-3 bg-white/90 hover:bg-white"
        >
          <Heart className="h-4 w-4 fill-red-500 text-red-500" />
        </Button>

        {/* View details button */}
        <Button
          size="sm"
          variant="ghost"
          className="absolute bottom-3 left-3 bg-white/90 hover:bg-white opacity-0 group-hover:opacity-100 transition-opacity"
        >
          <Eye className="h-4 w-4 text-gray-600" />
        </Button>
      </div>

      <CardHeader className="pb-3">
        <div className="flex justify-between items-start">
          <div className="space-y-1">
            <h3 className="font-bold text-lg leading-tight">{superheroName}</h3>
            <p className="text-sm text-gray-600">{name}</p>
          </div>
          <Badge className="text-xs bg-green-100 text-green-800 border-green-200">
            {type}
          </Badge>
        </div>
        <Badge variant="outline" className="w-fit text-xs">
          {league}
        </Badge>
      </CardHeader>

      <CardContent className="space-y-4">
        <p className="text-sm text-gray-600 line-clamp-2">{description}</p>

        {/* Stats */}
        <div className="grid grid-cols-2 gap-3">
          <div className="space-y-1">
            <div className="flex items-center gap-1">
              <Zap className="h-3 w-3 text-orange-500" />
              <span className="text-xs font-medium">Strength</span>
            </div>
            <Progress
              value={strength}
              className="h-2"
              activeColor="bg-orange-500"
            />
          </div>
          <div className="space-y-1">
            <div className="flex items-center gap-1">
              <Brain className="h-3 w-3 text-blue-500" />
              <span className="text-xs font-medium">Intelligence</span>
            </div>
            <Progress
              value={intelligence}
              className="h-2"
              activeColor="bg-blue-500"
            />
          </div>
          <div className="space-y-1">
            <div className="flex items-center gap-1">
              <Gauge className="h-3 w-3 text-green-500" />
              <span className="text-xs font-medium">Speed</span>
            </div>
            <Progress
              value={speed}
              className="h-2"
              activeColor="bg-green-500"
            />
          </div>
          <div className="space-y-1">
            <div className="flex items-center gap-1">
              <Shield className="h-3 w-3 text-purple-500" />
              <span className="text-xs font-medium">Durability</span>
            </div>
            <Progress
              value={durability}
              className="h-2"
              activeColor="bg-purple-500"
            />
          </div>
        </div>

        {/* Powers */}
        <div className="space-y-2">
          <h4 className="font-medium text-sm">Powers:</h4>
          <div className="flex flex-wrap gap-1">
            <Badge variant="outline" className="text-xs">
              Super Strength
            </Badge>
            <Badge variant="outline" className="text-xs">
              Flight
            </Badge>
            <Badge variant="outline" className="text-xs bg-gray-100">
              +4 more
            </Badge>
          </div>
        </div>

        <div className="text-xs text-gray-500 pt-2 border-t">
          First appeared: {firstAppeard}
        </div>
      </CardContent>
    </Card>
  );
};
