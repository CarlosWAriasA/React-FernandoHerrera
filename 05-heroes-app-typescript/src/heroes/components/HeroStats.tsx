import { Users, Heart, Zap, Trophy } from "lucide-react";
import { use } from "react";

import { Badge } from "@/components/ui/badge";
import { HeroStatCard } from "./HeroStatCard";
import { useHeroSummary } from "../hooks";
import { FavoriteHeroContext } from "../context/FavoriteHeroContext";

export const HeroStats = () => {
  const { data: summaryResponse } = useHeroSummary();
  const { favoriteCount } = use(FavoriteHeroContext);

  if (!summaryResponse) {
    return <div>Loading...</div>;
  }

  return (
    <div className="grid grid-cols-2 md:grid-cols-4 gap-4 mb-8">
      <HeroStatCard
        title="Total Characters"
        icon={<Users className="h-4 w-4 text-muted-foreground" />}
      >
        <div className="text-2xl font-bold">{summaryResponse?.totalHeroes}</div>
        <div className="flex gap-1 mt-2">
          <Badge variant="secondary" className="text-xs">
            {summaryResponse?.heroCount} Heroes
          </Badge>
          <Badge variant="destructive" className="text-xs">
            {summaryResponse?.villainCount} Villains
          </Badge>
        </div>
      </HeroStatCard>

      <HeroStatCard
        title="Favorites"
        icon={<Heart className="h-4 w-4 text-muted-foreground" />}
      >
        <div className="text-2xl font-bold text-red-600">{favoriteCount}</div>
        <p className="text-xs text-muted-foreground">
          {((favoriteCount / summaryResponse?.totalHeroes) * 100).toFixed(2)}%
          of total
        </p>
      </HeroStatCard>

      <HeroStatCard
        title="Strongest"
        icon={<Zap className="h-4 w-4 text-muted-foreground" />}
      >
        <div className="text-lg font-bold">
          {summaryResponse?.strongestHero?.alias}
        </div>
        <p className="text-xs text-muted-foreground">
          Strength: {summaryResponse?.strongestHero?.strength}/10
        </p>
      </HeroStatCard>

      <HeroStatCard
        title="Smartest"
        icon={<Trophy className="h-4 w-4 text-muted-foreground" />}
      >
        <div className="text-lg font-bold">
          {summaryResponse?.smartestHero?.alias}
        </div>
        <p className="text-xs text-muted-foreground">
          Intelligence: {summaryResponse?.smartestHero?.intelligence}/10
        </p>
      </HeroStatCard>
    </div>
  );
};
