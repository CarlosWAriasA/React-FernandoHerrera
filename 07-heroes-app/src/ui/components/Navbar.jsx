import { Link, NavLink } from "react-router";

export const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark p-2">
      <Link className="navbar-brand" to="/">
        Associations
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">
          <NavLink
            className={(args) =>
              `nav-item nav-link ${args.isActive ? "active" : ""}`
            }
            to="/marvel"
          >
            Marvel
          </NavLink>

          <NavLink
            className={(args) =>
              `nav-item nav-link ${args.isActive ? "active" : ""}`
            }
            to="/dc"
          >
            DC
          </NavLink>
        </div>
      </div>

      <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
        <ul className="navbar-nav ml-auto">
          <span className="nav-item nav-link text-primary">Carlos</span>
          <button className="nav-item nav-link btn">Logout</button>
        </ul>
      </div>
    </nav>
  );
};
