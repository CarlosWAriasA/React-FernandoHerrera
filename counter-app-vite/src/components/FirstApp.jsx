import React from "react";
import PropTypes from "prop-types";

const myName = "Carlos";

const Greeting = (name) => {
  return `Hey ${name}!`;
};

function FirstApp({ title, subTitle }) {
  return (
    <>
      <h1>{title}</h1>
      <p>{subTitle}</p>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  title: `There's not title`,
  subTitle: `There's not subtitle`,
};

export default FirstApp;
