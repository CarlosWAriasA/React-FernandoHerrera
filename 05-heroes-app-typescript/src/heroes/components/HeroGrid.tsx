import { HeroGridCard } from "./HeroGridCard";

export const HeroGrid = () => {
  return (
    <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8 mb-8">
      <HeroGridCard
        superheroName="Superman"
        name="Clark Kent"
        status="Active"
        universe="DC"
        league="Justice League"
        type="Hero"
        description="The Last Son of Krypton, protector of Earth and symbol of hope for
            all humanity."
        strength={100}
        intelligence={80}
        speed={90}
        durability={100}
        firstAppeard={1938}
      />
    </div>
  );
};
