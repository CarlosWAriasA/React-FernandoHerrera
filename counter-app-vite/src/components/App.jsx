import React from "react";
import FirstApp from "./FirstApp";
import CounterApp from "./CounterApp";

export default function App() {
  return (
    <>
      <FirstApp title="Carlos" />
    </>
  );
}
