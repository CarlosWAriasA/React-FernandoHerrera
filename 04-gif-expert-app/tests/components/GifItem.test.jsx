import { GifItem } from "../../src/components/GifItem";
import { render, screen } from "@testing-library/react";
import React from "react";

describe("Tests in <GifItem />", () => {
  const title = "One Punch";
  const url = "https://one-punch.com/img.jpg";

  test("should match with the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show the image with the URL and the ALT", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test("should have the title in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
