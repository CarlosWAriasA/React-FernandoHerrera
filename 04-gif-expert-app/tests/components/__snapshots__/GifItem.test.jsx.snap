// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Tests in <GifItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://one-punch.com/img.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
